parser/src/lib.rs

```diff
@@ -70,7 +70,7 @@ pub fn run_parser(
         let program = parser_logic::parse_file(&src, file_id)
             .map_err(|e| (file_library.clone(), vec![e]))?;
         if let Some(main) = program.main_component {
-            main_components.push((file_id, main));
+            main_components.push((file_id, main, program.custom_gates));
```

Comment:
We know if the main component is using the custom templates pragma by checking the field AST.custom_gates